README.md

```diff
@@ -0,0 +1,50 @@
+# koa-esm-transform
+
+Middleware for Koa servers that transforms standard JavaScript modules to earlier versions of JavaScript and/or AMD modules (inlining loader script `@polymer/esm-amd-loader` into the HTML), for use with older browsers.  
+
+Consider an HTML file containing the following inline JavaScript module:
+
+```html
+<script type="module">
+import {someFunction} from './some-module.js';
+someFunction();
+</script>
+```
+
+The koa-esm-to-amd middleware would transform that HTML code to something like the following:
+
+```html
+<script>
+// An inline loader script from `@polymer/esm-amd-loader` package
+// which adds the `define` function used below.
+</script>
+<script>
+define(["./some-module.js"], function (_someModule) {
+  "use strict";
+
+  (0, _someModule.someFunction)();
+});
+</script>
+```
+
+Because this is middleware, you can use it in a simple static file server as well with a proxy server.  This makes it possible to use in front of a test server such as the one `karma` starts up. (See [koa-karma-proxy](https://github.com/Polymer/koa-karma-proxy) for examples.)
+
+Note: HTML and JavaScript are parsed on every request for those content-types, it is intended for use in development context to facilitate build-free testing/iteration as opposed to in a high volume production web server.
+
+## How it works
+
+By default, a set of babel transform plugins are chosen based on the known capabilities of the browser/user-agent identified in the Koa Context for the request.
+
+When downstream server returns HTML content, it is scanned for module script tags (e.g. `<script type="module">`).  Any `src` attribute values are appended with the `__esmTransform` queryparameter.
```

Comment:
Suggest adding something like: "We add the query parameter because we need to distinguish between scripts that are being imported as a module vs a traditional script -- something we can't necessarily tell by looking at the script itself." (assuming I understand it correctly!)

Reply:
Thx!

I went with

> The query parameter is added because the middleware needs to distinguish between scripts that are being imported as a module vs a traditional script and this can't necessarily be determined by looking at the script itself.